src/Utils/get.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Type-safe access of deep property of an object
+ *
+ * Thanks! https://github.com/claudepache/es-optional-chaining/issues/2#issuecomment-318680932
+ *
+ * @param obj                     Object to get deep property
+ * @param unsafeDataOperation     Function that returns the deep property
+ * @param valueIfFailOrUndefined  Value to return in case if there is no such property
+ */
+export function get<O, T>(
+  obj: O,
```

Comment:
I wonder if the input object is even really necessary? We really only care about `T` and can’t we have that by doing just `get(() => obj.deeply.nested.access ))`?

Reply:
That's true I guess... it does somewhat enforce that you understand what object you're acting on though. A bit more explicit, but that's not always a bad thing.

Reply:
`get(() => obj.deeply.nested.access ))` is interesting but I think the input / output approach maps well with other common implementations.

Reply:
Sounds good 👍